jflex/src/test/java/jflex/state/StateSetQuickcheck.java

```diff
@@ -208,6 +238,12 @@ public void complementElements(StateSet s1, StateSet s2) {
     StateSet comp = s1.complement(s2);
     // only elements of s2 are in the complement
     assertThat(s2.contains(comp)).isTrue();
+
+    // ensure that comp does not contain s1
+    StateSet empty = new StateSet();
+    if (!empty.contains(s1)) { // if s1 is {}, then it will always be contained in comp
```

Comment:
prefer `s1.containsElements()` instead of comparison with a new empty set

Reply:
I don't remember if I have that property already, but a separate property for `!empty.contains(set)` and `!empty.containsElements()` as well as other algebraic properties of the empty set would be a nice addition in general.